opencog/moses/main/demo-problems.cc

```diff
@@ -261,20 +261,20 @@ void polynomial_problem::run(option_base* ob)
 	if (pms.exemplars.empty()) {
 		pms.exemplars.push_back(type_to_exemplar(id::contin_type));
 	}
+	type_tree tt = gen_signature(id::contin_type, 1);
+	ITable it(tt, (pms.nsamples>0 ? pms.nsamples : pms.default_nsamples));
+	type_node_seq tt_seq = type_tree_to_tyn_seq(tt);
 
 	if (pms.deme_params.atomspace_port) {
 		// atomspace used for storing candidate programs
 		pms.deme_params.as = new AtomSpace();
+		populate(pms.deme_params.as, it,tt_seq);
```

Comment:
Rather than changing your implementation on `populate` and adding additional optional parameter, have you tried just setting the type if the input table using `ITable::set_types(const type_node_seq &il)`.

Reply:
I haven't but both ways are possible.

Reply:
then I suggest using set_types for three reasons. The process for demo and table based problems will be the same which makes it easier to understand and maintain the code, You wouldn't have to add more code to populate and last the type can be reused if it is in ITable avoiding regenerating/inferring the type.